### ISSUE TITLE
Block autobuild on server?

### ISSUE THREAD

**Issue by HyperjumpX** [Lunatrius/Schematica#426]
I have a server with about 100 players. some of them want to use Schematica.
Schematica itself is great but i want the players to build by hand not by autobuild.
Is there a way to allow the ghostimages and block autobuild?

**Comment by Pokechu22** [comment]
Yes; look into the schematica plugin (I don't remember the exact name of it).  There's a plugin message system you can use to enable/disable the printer, among other things.

**Comment by HyperjumpX** [comment]
oh nice,
sry i need more details. I have just claimed the rights for the server and digging through java right now :I

In plugin configuration clientside is a surface for allowing printer or disable it.
Is that what u ment?

Because i have to force the players by serverside to block it :I 
I hope i didnt misunderstood you :O

**Comment by BlueAgent** [comment]
If you're running a forge server you can drop the [Schematica](https://minecraft.curseforge.com/projects/schematica) mod on to the server, and adjust the following configs:
```
server {
    # Allow players to load schematics.
    B:loadEnabled=true

    # Amount of storage provided per-player for schematics on the server.
    I:playerQuotaKilobytes=8192

    # Allow players to use the printer.
    B:printerEnabled=true 

    # Allow players to save schematics.
    B:saveEnabled=true
}
```

If you're running a Spigot server there's [SchematicaPlugin](https://www.spigotmc.org/resources/schematicaplugin.14411/) that allows you to adjust usage of the printer, saving and loading features with these permission nodes:
```
schematica.printer
schematica.save
schematica.load
```

As for Sponge servers...
Not sure if a plugin exists for SpongeVanilla, but you could just use the mod itself for SpongeForge.

Of course people who are determined enough can patch the printer disabling feature out of Schematica or even use [Baritone](https://github.com/cabaletta/baritone) as an alternative to the Schematica printer, which also probably gets past most Anti-Cheat, at least for now.

**Comment by HyperjumpX** [comment]
Thank you very much. That hopefully gives plenty of happy faces ;)

**Comment by LoganDark** [comment]
> If you're running a vanilla server, there's a spigot [snip]

No, Spigot isn't vanilla.

**Comment by BlueAgent** [comment]
> > If you're running a vanilla server, there's a spigot [snip]
> 
> No, Spigot isn't vanilla.

MB. Fixed.
